### PR TITLE
Refactoring _get_plot_stat_map_params

### PR DESCRIPTION
**Pull request by juhuntenburg**
Refers to #699. 
Adding tests to show that the function fails and fixing function to make it pass the tests. Try to make the function more easily readable.

### CODE REVIEW

**Comment by bthirion** [comment]
I have not noticed any big flaw.

File: nilearn/plotting/tests/test_img_plotting.py [hunk 1]
```diff
@@ -445,3 +446,171 @@ def test_plot_prob_atlas():
...
+def test_get_plot_stat_map_params_pos_neg():
+    # data with positive and negative range
+    affine = np.eye(4)
+    data = np.array([[-.5, 1., 0.],
+                     [0., 1., -.2],
+                     [1.5, 2.5, 3.]])
+    img = nibabel.Nifti1Image(data, affine)
+
+    # symmtric_cbar set to True
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_plot_stat_map_params(img, None,
```
Comment (lesteve) [inline nilearn/plotting/tests/test_img_plotting.py]: Just for clarity, it'd be quite good to name the parameters in the _get_plot_stat_map_params function, i.e.:

```python
_get_plot_stat_map_params(img, vmax=2, symmetric_cbar=True, kwargs={})
```

I find myself always going back to the function signature ...

**Comment by juhuntenburg** [comment]
Turns out the logic was correct in the beginning and I was only confused by the doc string of plot_stat_map / plot_glass_brain. So I will simply improve the doc string

**Comment by lesteve** [comment]
You should definitely reuse and update your tests, though. This colorbar code has given us headaches in the past and it'd be great to have it tested more thoroughly.

**Comment by eickenberg** [comment]
> You should definitely reuse and update your tests, though. This colorbar
code has given us headaches in the past and it'd be great to have it tested
more thoroughly.

+ 1e384

File: nilearn/plotting/img_plotting.py [hunk 2]
```diff
@@ -43,27 +43,38 @@
...
     # Avoid dealing with masked_array:
-        if hasattr(stat_map_data, '_mask'):
-            stat_map_data = np.asarray(stat_map_data[np.logical_not(stat_map_data._mask)])
-        stat_map_max = np.nanmax(stat_map_data)
-        if force_min_stat_map_value == None:
-            stat_map_min = np.nanmin(stat_map_data)
-        else:
-            stat_map_min = force_min_stat_map_value
+    if hasattr(stat_map_data, '_mask'):
+        stat_map_data = \
```
Comment (banilo) [inline nilearn/plotting/img_plotting.py]: I think backslash is bad style and to be replaced by brackets,..

Comment (AlexandreAbraham) [inline nilearn/plotting/img_plotting.py]: In that case, there is this option:

```python
        stat_map_data = np.asarray(
            stat_map_data[np.logical_not(stat_map_data._mask)])

**Pull request closed without merging**